nomic/aws/sagemaker.py

```diff
@@ -0,0 +1,197 @@
+    nomic_embed_text_v1_5 = "nomic-embed-text-v1.5"
+
+    def hamming_capable(self):
+        return self == NomicTextEmbeddingModel.nomic_embed_text_v1_5
+
+    def matryoshka_capable(self):
+        return self == NomicTextEmbeddingModel.nomic_embed_text_v1_5
+
+    def dim(self):
+        return text_embedding_model_info[self.value]["dim"]
+
+    def recommended_dims(self):
+        return text_embedding_model_info[self.value].get("recommended_dims") or [
+            self.dim()
+        ]
+
+    def max_length(self):
+        return text_embedding_model_info[self.value]["max_length"]
+
+    def pad_token(self) -> int:
+        return text_embedding_model_info[self.value]["pad_id"]
+
+
+def null_empty_placeholder(text: Optional[str]) -> str:
+    """
+    If text is null or empty replace with a placeholder nonsense string.
+    """
+    if text is None:
+        return NULL_PLACEHOLDER
+    if text.strip() == "":
+        return EMPTY_PLACEHOLDER
+    return text
+
+
+def load_tokenizer(model: NomicTextEmbeddingModel) -> Tokenizer:
+    """
+    Instantiate the huggingface tokenizer.
+    :param model: the model doing the embedding
+    :return:
+    """
+    tokdir = Path(__file__).parent / "tokenizers"
+    if model == NomicTextEmbeddingModel.nomic_embed_text_v1_5:
+        tokenizer = Tokenizer.from_file(str(tokdir / "bert-base-uncased.json"))
+        # TODO: change to 8192
+        tokenizer.enable_truncation(max_length=2048)
+        tokenizer.enable_padding(pad_id=0, pad_token="[PAD]", pad_to_multiple_of=64)
+    else:
+        raise Exception(f"Could not determine tokenizer for model: {model.value}")
+    return tokenizer
+
+
+def create_sagemaker_request_for_batch(texts: List[str], tokenizer: Tokenizer):
+    """
+    Tokenizes and creates a Triton embedding request from a list of texts.
+
+    Args:
+        texts: List of texts to be batched.
+        tokenizer: Tokenizer to use.
+
+    Returns:
+        HTTP Request object for Triton server.
+    """
+    encoded = tokenizer.encode_batch(texts)
+
+    input_ids = np.array([e.ids for e in encoded], dtype=np.int32)
+    attention_mask = np.array([e.attention_mask for e in encoded], dtype=np.int32)
+
+    inputs = []
+    outputs = []
+
+    inputs.append(aiohttpclient.InferInput("input_ids", list(input_ids.shape), "INT32"))
+    inputs.append(
+        aiohttpclient.InferInput("attention_mask", list(input_ids.shape), "INT32")
+    )
+
+    # Initialize the data
+    inputs[0].set_data_from_numpy(input_ids, binary_data=True)
+    inputs[1].set_data_from_numpy(attention_mask, binary_data=True)
+
+    # have to set to binary since http doesn't natively support fp16
+    outputs.append(aiohttpclient.InferRequestedOutput("embedding", binary_data=True))
+    request, header_length = InferenceServerClient.generate_request_body(
+        inputs=inputs, outputs=outputs
+    )
+    return request, header_length
+
+
+def batch_sagemaker_requests(texts: List[str], batch_size=32):
```

Comment:
make sure we test the longest text length we'll allow with the batch size we specify here

Reply:
i thought we truncated the tokens?